benches/src/erc721.rs

```diff
@@ -0,0 +1,118 @@
+use alloy::{
+    network::{AnyNetwork, EthereumWallet},
+    primitives::Address,
+    providers::ProviderBuilder,
+    sol,
+    sol_types::SolConstructor,
+    uint,
+};
+use e2e::{receipt, Account};
+
+use crate::ArbOtherFields;
+
+sol!(
+    #[sol(rpc)]
+    contract Erc721 {
+        function balanceOf(address owner) external view returns (uint256 balance);
+        function approve(address to, uint256 tokenId) external;
+        function getApproved(uint256 tokenId) external view returns (address approved);
+        function isApprovedForAll(address owner, address operator) external view returns (bool approved);
+        function ownerOf(uint256 tokenId) external view returns (address ownerOf);
+        function safeTransferFrom(address from, address to, uint256 tokenId) external;
+        function setApprovalForAll(address operator, bool approved) external;
+        function totalSupply() external view returns (uint256 totalSupply);
+        function transferFrom(address from, address to, uint256 tokenId) external;
+        function mint(address to, uint256 tokenId) external;
+        function burn(uint256 tokenId) external;
+    }
+);
+
+sol!("../examples/erc721/src/constructor.sol");
+
+pub async fn bench() -> eyre::Result<()> {
+    let alice = Account::new().await?;
+    let alice_addr = alice.address();
+    let alice_wallet = ProviderBuilder::new()
+        .network::<AnyNetwork>()
+        .with_recommended_fillers()
+        .wallet(EthereumWallet::from(alice.signer.clone()))
+        .on_http(alice.url().parse()?);
+
+    let bob = Account::new().await?;
+    let bob_addr = bob.address();
+
+    let contract_addr = deploy(&alice).await;
+    let contract = Erc721::new(contract_addr, &alice_wallet);
+
+    let _ = receipt!(contract.mint(alice_addr, uint!(2_U256)))?;
```

Comment:
Nit: I would create `let token_2 = uint!(2_U256);` and then reuse it.

Reply:
Yeah, this sounds better